library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionParameters.java

```diff
@@ -106,11 +208,45 @@ public SectionParameters build() {
     }
 
     private SectionParameters(Builder builder) {
+        this.itemResourceId = builder.itemResourceId;
         this.headerResourceId = builder.headerResourceId;
         this.footerResourceId = builder.footerResourceId;
-        this.itemResourceId = builder.itemResourceId;
         this.loadingResourceId = builder.loadingResourceId;
         this.failedResourceId = builder.failedResourceId;
         this.emptyResourceId = builder.emptyResourceId;
+        this.itemViewWillBeProvided = builder.itemViewWillBeProvided;
+        this.headerViewWillBeProvided = builder.headerViewWillBeProvided;
+        this.footerViewWillBeProvided = builder.footerViewWillBeProvided;
+        this.loadingViewWillBeProvided = builder.loadingViewWillBeProvided;
+        this.failedViewWillBeProvided = builder.failedViewWillBeProvided;
+        this.emptyViewWillBeProvided = builder.emptyViewWillBeProvided;
+
+        if (itemResourceId != null && itemViewWillBeProvided) {
```

Comment:
could you please add unit tests to this class to validate these checks?

Reply:
Done. Added SectionsParametersTest.